### PR TITLE
chore(medusa): clean up swap creation to use the new calculation

### PR DESCRIPTION
**Pull request by adrien2p**
The scope of that issue has been changed to a boy scout ticket instead. it is now only including boy scouting clean up and improvements

FIXES CORE-1085

### CODE REVIEW

File: packages/medusa/src/services/swap.ts [hunk 1]
```diff
@@ -1221,6 +1216,33 @@ class SwapService extends TransactionBaseService {
       return result
     })
   }
+
+  protected async areReturnItemsValid(
+    returnItems: WithRequiredProperty<Partial<ReturnItem>, "item_id">[]
+  ): Promise<boolean> {
+    const manager = this.transactionManager_ ?? this.manager_
```
Comment (olivermrbl) [inline packages/medusa/src/services/swap.ts:1223]: thought: We've done this many places by now. Do you think it would make sense to create a method in the parent base service that always retrieves the correct manager?

I noticed @carlos-r-l-rodrigues added it in a separate class:
```
getManager() {
  return this.transactionManager_ ?? this.manager_
}
```

Comment (adrien2p) [reply packages/medusa/src/services/swap.ts:1223]: Let me know what you think

Comment (olivermrbl) [reply packages/medusa/src/services/swap.ts:1223]: Sure, I think that's really neat 👍

**Review by fPolic — APPROVED** [review]
LGTM

File: packages/medusa/src/interfaces/transaction-base-service.ts [hunk 2]
```diff
@@ -2,14 +2,20 @@ import { EntityManager } from "typeorm"
...
+    return this.transactionManager_ ?? this.manager_
+  }
 
   protected constructor(
     protected readonly __container__: any,
     protected readonly __configModule__?: Record<string, unknown>,
     protected readonly __moduleDeclaration__?: Record<string, unknown>
-  ) {}
+  ) {
+    this.manager_ = __container__.manager
```
Comment (olivermrbl) [inline packages/medusa/src/interfaces/transaction-base-service.ts:17]: question: Can this have unexpected side effects? Seems to be unrelated to the general boy scouting in the PR

Comment (adrien2p) [reply packages/medusa/src/interfaces/transaction-base-service.ts:17]: it should not have unexpected side effect since that if you want to use the TransctionBaseService the manager should be present. The transactionBaseService itself use the manager_ so it expect it to be present. Also, the property is not optional so it has to be populated.

**Review by olivermrbl — APPROVED** [review]
LGTM!
